### PR TITLE
build Mixxx with macOS 10.13 SDK

### PR DESCRIPTION
**Pull request by Be-ing**
to work around performance issues with newer SDKs

### CODE REVIEW

**Comment by Holzhaus** [comment]
What about the buildenv? Is that also built with the old SDK?

**Comment by Be-ing** [comment]
No, but I don't think that's necessary.

**Comment by Be-ing** [comment]
If rebuilding the dependencies turns out to be necessary, that may be tricky with Qt's build system.

**Comment by Be-ing** [comment]
Performance is no better with this build. Let's try with the old build environment...

**Comment by Be-ing** [comment]
Actually, the first build did not really use the macOS 10.13 SDK because of a typo in the macos_buildenv.sh script. Testing this again with a build that did actually use the 10.13 SDK, performance is good. So I think we should go with this hack. I would like to add Ninja to the build environment so the macos_buildenv.sh script doesn't need to download it separately.

However, performance is bad in LateNight with the level meters showing. Other skins work okay even with level meters, spinnies, and OpenGL waveforms. @ronso0 do you have any ideas why the meters in LateNight might be performing worse? They are PNGs...

**Comment by Be-ing** [comment]
Interesting, apparently the Unix Makefiles generator works and setting CC and CXX environment variables are not necessary on GitHub Actions, but they are on the old Macbook I'm testing with macOS 10.13.

**Comment by Be-ing** [comment]
I tested building on macOS 11, but Apple just loves to break things:

**Comment by Be-ing** [comment]
If we're lucky this might fix the PortAudio issues on macOS 11 too.

**Comment by Be-ing** [comment]
On further testing, no, performance isn't better with this. But this at least shows that it is possible to use an old macOS SDK on GitHub Actions. I do think the build environment needs to be rebuilt. Now that we have the macos_buildenv.sh script, this shouldn't be an obstacle to macOS developers building locally.

File: tools/macos_buildenv.sh [hunk 1]
```diff
@@ -78,12 +78,39 @@ case "$COMMAND" in
             echo "Build environment found: ${BUILDENV_PATH}"
         fi
 
+        export SDKROOT="${BUILDENV_BASEPATH}/MacOSX10.13.sdk"
```
Comment (uklotzde) [inline tools/macos_buildenv.sh:81]: Shall we define a target SDK that is by default set to the current SDK? Then we only need to go the extra mile of downloading it and manipulating env variables if it doesn't match the native SDK.

Comment (Be-ing) [reply tools/macos_buildenv.sh:81]: No, practically nobody is using a version of XCode that has such an old macOS SDK anymore.

**Comment by Be-ing** [comment]
> I tested building on macOS 11, but Apple just loves to break things...

This has already been fixed upstream in CMake 3.19.

**Comment by Be-ing** [comment]
We have confirmation that just building Mixxx with the macOS 10.13 SDK fixes the performance problems. No need to revert to the old build environment or rebuild the build environment with the 10.13 SDK. Ready for merge.

**Comment by Be-ing** [comment]
ping

**Review by Holzhaus — APPROVED** [review]
I can't test this obviously, but code looks okay. Thank you for looking into these issues.

**Pull request merged**